mautic/mautic
# Fix Salesforce sync boolean type value

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for features or enhancements / 3.2 for bug fixes 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
We've noticed this issue:

> Impossible of unserialize boolean instance of VALUEè_STRING value 1 or a field is missing in the request at [line:1, column:2]

Boolean type parameters was converted to string, because amendLeadDataBeforePush use str_replace for all type of parameters and convert it to string. 
This PR fixed it



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Code review should be enough, because setup integration is not easy.
3. Check unit test
